src/naive/datetime/mod.rs

```diff
@@ -533,6 +533,34 @@ impl NaiveDateTime {
         Some(NaiveDateTime { date, time })
     }
 
+    /// Adds given `Months` to the current date and time.
+    ///
+    /// Returns `None` when it will result in overflow.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use std::str::FromStr;
+    /// use chrono::{Months, NaiveDateTime};
+    /// let dt = |s| NaiveDateTime::from_str(s).unwrap();
+    /// let ca = |dt: NaiveDateTime, m: Months| dt.checked_add_months(m).unwrap();
+    /// let m = Months::new;
+    /// assert_eq!(ca(dt("2014-01-01T01:00:00"), m(1)), dt("2014-02-01T01:00:00"));
```

Comment:
I'm a little uncomfortable with this style of code (albeit similar to what is used in the private tests) being in the public docs. IMO this would look better if we just used the full functions directly, with more intermediate variables if needed for readability. 

I've set up https://github.com/avantgardnerio/chrono/pull/1 as one possibility here, but I'm not set on that option, and keen to have a discussion on the merits of each approach.

Reply:
I agree that this makes the examples harder to understand (and somewhat unidiomatic).

Reply:
I think they are all fixed now. Please ping me if I missed any!